customer-service/src/main/scala/Server.scala

```diff
@@ -1,39 +1,47 @@
 import cats.effect._
 import cats.implicits.toBifunctorOps
-import domain.CustomerService
+import domain.{CustomerService, HttpClient}
 import gateway.CustomerRoutes.customerRoutes
 import infrastructure.repositories.InMemoryCustomerRepository
 import infrastructure.repositories.config.ServerConf
-import org.http4s.HttpApp
 import org.http4s.blaze.server.BlazeServerBuilder
-import org.http4s.server.Router
 import org.http4s.implicits._
+import org.http4s.server.Router
 import pureconfig.ConfigSource
 import pureconfig.generic.auto._
 
 object Server extends IOApp {
 
-  private val customerRepo = new InMemoryCustomerRepository[IO]
-  val customerService      = new CustomerService[IO](customerRepo)
-
-  private val httpApp: HttpApp[IO] = Router(
-    "/api" -> customerRoutes[IO](customerService)
-  ).orNotFound
-
   private def loadConfig: IO[ServerConf] =
     IO.fromEither(
       ConfigSource.default.load[ServerConf].leftMap(error => new RuntimeException(s"Failed to load config $error"))
     )
 
-  override def run(args: List[String]): IO[ExitCode] =
-    loadConfig.flatMap { serverConfig =>
-      BlazeServerBuilder[IO]
+  private def buildCustomerService: CustomerService[IO] = {
+    val client       = HttpClient[IO](8080, "127.0.0.1")
+    val customerRepo = new InMemoryCustomerRepository[IO]
+    new CustomerService[IO](customerRepo, client)
+  }
+
+  override def run(args: List[String]): IO[ExitCode] = {
+
+    for {
+      serverConfig    <- loadConfig
+      customerService <- IO.pure(buildCustomerService)
```

Comment:
don't wrap in effect type on the caller side, try to always return effect types from callee

nearly never: `IO.pure(X.y())`, nearly always`X { def y(): IO[...] }`